tests/unit/test_detect_target_contour.py

```diff
@@ -0,0 +1,286 @@
+    for i in range(0, len(expected.detections)):
+        expected_detection = expected.detections[i]
+        actual_detection = actual.detections[i]
+
+        assert expected_detection.label == actual_detection.label
+        np.testing.assert_almost_equal(
+            expected_detection.confidence,
+            actual_detection.confidence,
+            decimal=CONFIDENCE_PRECISION_TOLERANCE,
+        )
+
+        np.testing.assert_almost_equal(
+            actual_detection.x_1,
+            expected_detection.x_1,
+            decimal=BOUNDING_BOX_PRECISION_TOLERANCE,
+        )
+
+        np.testing.assert_almost_equal(
+            actual_detection.y_1,
+            expected_detection.y_1,
+            decimal=BOUNDING_BOX_PRECISION_TOLERANCE,
+        )
+
+        np.testing.assert_almost_equal(
+            actual_detection.x_2,
+            expected_detection.x_2,
+            decimal=BOUNDING_BOX_PRECISION_TOLERANCE,
+        )
+
+        np.testing.assert_almost_equal(
+            actual_detection.y_2,
+            expected_detection.y_2,
+            decimal=BOUNDING_BOX_PRECISION_TOLERANCE,
+        )
+
+
+def create_detections(detections_from_file: np.ndarray) -> detections_and_time.DetectionsAndTime:
+    """
+    Create DetectionsAndTime from expected.
+    Format: [confidence, label, x_1, y_1, x_2, y_2] .
+    """
+    assert detections_from_file.shape[1] == 6
+
+    result, detections = detections_and_time.DetectionsAndTime.create(0)
+    assert result
+    assert detections is not None
+
+    for i in range(0, detections_from_file.shape[0]):
+        result, detection = detections_and_time.Detection.create(
+            detections_from_file[i][2:],
+            int(detections_from_file[i][1]),
+            detections_from_file[i][0],
+        )
+        assert result
+        assert detection is not None
+        detections.append(detection)
+
+    return detections
+
+
+# ------------------------------------------------------------------------------------------------------------------
+@pytest.fixture()
+def detector() -> detect_target_contour.DetectTargetContour:  # type: ignore
+    """
+    Construct DetectTargetUltralytics.
+    """
+    detection = detect_target_contour.DetectTargetContour()
+    yield detection  # type: ignore
+
+
+# ---------------------------------------------------------------------------
+class TestDetector:
+    """
+    Tests `DetectTarget.run()` .
+    """
+
+    def test_multiple_landing_pads(
+        self,
+        detector: detect_target_contour.DetectTargetContour,
+    ) -> None:
+        """
+        Multiple images.
+        """
+
+        circle_data = [
+            [(200, 200), 400, False, [False, None, None]],
+            [(1500, 700), 500, False, [False, None, None]],
+        ]
+
+        actual_detections, expected_detections = [], []
+        circle_list = [circle_data]
+
+        for i, circle_data in enumerate(circle_list):
```

Comment:
Do not use a loop. Each unit test tests 1 image (that's why it's called a *unit* test).

There is a loop in one of the YOLOv8 unit test because it's a black box, so that test is to confirm that the model is stateless. The contour code written by Autonomy is already known to be stateless, so replicating that test here is not useful.